h5netcdf/core.py

```diff
@@ -578,14 +592,31 @@ def _check_dtype(self, dtype):
         # check if committed type can be accessed in current group hierarchy
         dname = dtype.name.split("/")[-1]
         if (
-            (user_type := self._all_enumtypes.get(dname)) is None
-        ) or user_type._h5ds.name != h5type.name:
+            (user_type := self._all_usertypes.get(dname)) is None
+        ) or self._root._h5file[user_type].name != h5type.name:
             msg = (
                 f"Given dtype {dtype.name!r} is not accessible in current group"
                 f" {self._h5group.name!r} or any parent group. Instead it's defined at"
                 f" {h5type.name!r}. Please create it in the current or any parent group."
             )
             raise TypeError(msg)
+    elif np.dtype(dtype).kind == "c":
+        itemsize = np.dtype(dtype).itemsize
+        try:
+            width = {8: "FLOAT", 16: "DOUBLE"}[itemsize]
+        except KeyError as e:
+            raise TypeError(
+                "Currently only 'complex64' and 'complex128' dtypes are allowed."
+            ) from e
+        dname = f"_PFNC_{width}_COMPLEX_TYPE"
+        # todo check compound type for existing complex types
+        #  which may be used her
+        # if dname is not available in current group-path
+        # create and commit type in current group
+        if dname not in self._all_cmptypes:
+            dtype = self.create_cmptype(dtype, dname).dtype
+        # get committed type from file
+        # return self._all_cmptypes[dname]._h5ds
```

Comment:
This is the complex compound creation part. This is really backwards compatible, as h5netcdf always reads these complex compound types as complex numbers. netcdf-c/netcdf4-python can read those compounds if, like we do now, the complex compound type is committed to the file.